backend/onedrive/onedrive.go

```diff
@@ -1079,6 +1079,8 @@ func (o *Object) Open(options ...fs.OpenOption) (in io.ReadCloser, err error) {
 	if err != nil {
 		return nil, err
 	}
+	//Overwrite size with actual size since size readings from Onedrive is unreliable.
+	o.size = resp.ContentLength
```

Comment:
According to [the docs](https://golang.org/pkg/net/http/#Response) ContentLength can be -1 if the size isn't known, so this shouldn't override the size in that case.